SpeechResponder/Service/ScriptRecoveryService.cs

```diff
@@ -0,0 +1,128 @@
+﻿using System;
+using System.IO;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+
+namespace EddiSpeechResponder.Service
+{
+	public class ScriptRecoveryService
+	{
+		static ScriptRecoveryService()
+		{
+			WorkingDirectory = Utilities.Constants.DATA_DIR;
+		}
+
+		public ScriptRecoveryService(EditScriptWindow scriptWindow)
+		{
+			_scriptWindow = scriptWindow;
+			_lockRoot = new object();
+		}
+
+		private static readonly string WorkingDirectory;
+		private string _tempFileName;
+		private EditScriptWindow _scriptWindow;
+		private bool _scriptSaveCallGuard;
+		private readonly object _lockRoot;
+
+		public static Script GetRecoveredScript()
+		{
+			var recoveredScriptPath = Path.Combine(WorkingDirectory, "editedScript.temp");
+
+			if (!File.Exists(recoveredScriptPath))
+			{
+				return null;
+			}
+
+			var recoveringScript = File.ReadAllText(recoveredScriptPath);
+			if (string.IsNullOrWhiteSpace(recoveringScript))
+			{
+				return null;
+			}
+
+			return JsonConvert.DeserializeObject<Script>(File.ReadAllText(recoveringScript));
```

Comment:
It looks to me like line 42 should be simplified to:
```cs
 			return JsonConvert.DeserializeObject<Script>(recoveringScript); 
```
There is no need to invoke `File.ReadAllText()` a second time - the recovered json string is not the path.